### ISSUE TITLE
Number of animals in the refinement tracklets step  does not coincide with the number of animals specified in the GUI

### ISSUE THREAD

**Issue by ana-ich** [DeepLabCut/DeepLabCut#1288]
OS: Windows 10
DeepLabCut Version: 2.2rc1
Anaconda env used: DLC-GPU

Describe the problem:

I'm using the GUI with the new version of DLC (I have a single animal) and in the Refine Tracklets step, even if the number of animals is specified 1, when I refine tracklets I have 2 individuals popping out (at some point the detections change color, detecting 2 animals on the same animal) 

**Screenshots**
![image](https://user-images.githubusercontent.com/66062281/120196749-e1efc680-c228-11eb-8504-a86afa375edf.png)
![image](https://user-images.githubusercontent.com/66062281/120196775-e9af6b00-c228-11eb-989b-ffad3974cf26.png)
![image](https://user-images.githubusercontent.com/66062281/120197160-5c204b00-c229-11eb-8bd5-05146011456c.png)
![image](https://user-images.githubusercontent.com/66062281/120197240-6c382a80-c229-11eb-832e-3b7380fd4882.png)

I want to add that I also have problems in the plotting section. I have a Key error in pandas.

Traceback (most recent call last):
  File "C:\Users\ephys\anaconda3\envs\DLC-GPU\lib\site-packages\pandas\core\indexes\base.py", line 3080, in get_loc
    return self._engine.get_loc(casted_key)
  File "pandas\_libs\index.pyx", line 70, in pandas._libs.index.IndexEngine.get_loc
  File "pandas\_libs\index.pyx", line 101, in pandas._libs.index.IndexEngine.get_loc
  File "pandas\_libs\hashtable_class_helper.pxi", line 4554, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas\_libs\hashtable_class_helper.pxi", line 4562, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'mouseDserine'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\ephys\anaconda3\envs\DLC-GPU\lib\site-packages\deeplabcut\gui\create_videos.py", line 363, in create_videos
    track_method=self.trackertypes.GetValue(),
  File "C:\Users\ephys\anaconda3\envs\DLC-GPU\lib\site-packages\deeplabcut\utils\plotting.py", line 279, in plot_trajectories
    linewidth=linewidth,
  File "C:\Users\ephys\anaconda3\envs\DLC-GPU\lib\site-packages\deeplabcut\utils\plotting.py", line 51, in PlottingResults
    Dataframe = Dataframe.loc(axis=1)[:, individuals2plot]
  File "C:\Users\ephys\anaconda3\envs\DLC-GPU\lib\site-packages\pandas\core\indexing.py", line 889, in __getitem__
    return self._getitem_tuple(key)
  File "C:\Users\ephys\anaconda3\envs\DLC-GPU\lib\site-packages\pandas\core\indexing.py", line 1060, in _getitem_tuple
    return self._getitem_lowerdim(tup)
  File "C:\Users\ephys\anaconda3\envs\DLC-GPU\lib\site-packages\pandas\core\indexing.py", line 787, in _getitem_lowerdim
    return self._getitem_axis(tup, axis=axis)
  File "C:\Users\ephys\anaconda3\envs\DLC-GPU\lib\site-packages\pandas\core\indexing.py", line 1117, in _getitem_axis
    locs = labels.get_locs(key)
  File "C:\Users\ephys\anaconda3\envs\DLC-GPU\lib\site-packages\pandas\core\indexes\multi.py", line 3305, in get_locs
    self.get_loc_level(k, level=i, drop_level=False)[0]
  File "C:\Users\ephys\anaconda3\envs\DLC-GPU\lib\site-packages\pandas\core\indexes\multi.py", line 2965, in get_loc_level
    return self._get_loc_level(key, level=level, drop_level=drop_level)
  File "C:\Users\ephys\anaconda3\envs\DLC-GPU\lib\site-packages\pandas\core\indexes\multi.py", line 3068, in _get_loc_level
    indexer = self._get_level_indexer(key, level=level)
  File "C:\Users\ephys\anaconda3\envs\DLC-GPU\lib\site-packages\pandas\core\indexes\multi.py", line 3158, in _get_level_indexer
    idx = self._get_loc_single_level_index(level_index, key)
  File "C:\Users\ephys\anaconda3\envs\DLC-GPU\lib\site-packages\pandas\core\indexes\multi.py", line 2809, in _get_loc_single_level_index
    return level_index.get_loc(key)
  File "C:\Users\ephys\anaconda3\envs\DLC-GPU\lib\site-packages\pandas\core\indexes\base.py", line 3082, in get_loc
    raise KeyError(key) from err
KeyError: 'mouseDserine'

'mouseDserine' being the only individual in the config file.

 Thank you for the help!

**Comment by MMathisLab** [comment]
You will have to update to rc2; 1 did not support single animal mode.

**Comment by ana-ich** [comment]
Actually I just checked and I have the rc2 version. 
![image](https://user-images.githubusercontent.com/66062281/120223465-50914c00-c24a-11eb-8cdc-19ed8d95276b.png)

**Comment by backyardbiomech** [comment]
I cannot reproduce the error in a maDLC project on a video with a single animal. However, I don't have any `uniquebodyparts` in this project, so that could be differences.

However, I cannot set the number of animals in the GUI to 1 (I just pulled from the repo now). If I type in 1, and then click `Create Tracks` button, it changes back to 2 and I get an expected error that it can't create 2 sets of tracks from the one animal in the video.

<details>
<summary> Traceback </summary>

```
100%|████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<?, ?it/s]
C:\Users\jacksonbe3\Miniconda3\envs\dlc22\lib\site-packages\deeplabcut\refine_training_dataset\stitch.py:650: UserWarning: No optimal solution found. Employing black magic...
  warnings.warn("No optimal solution found. Employing black magic...")
Traceback (most recent call last):
  File "C:\Users\jacksonbe3\Miniconda3\envs\dlc22\lib\site-packages\pandas\core\internals\managers.py", line 1676, in create_block_manager_from_blocks
    values=blocks[0], placement=slice(0, len(axes[0])), ndim=2
  File "C:\Users\jacksonbe3\Miniconda3\envs\dlc22\lib\site-packages\pandas\core\internals\blocks.py", line 2742, in make_block
    return klass(values, ndim=ndim, placement=placement)
  File "C:\Users\jacksonbe3\Miniconda3\envs\dlc22\lib\site-packages\pandas\core\internals\blocks.py", line 143, in __init__
    f"Wrong number of items passed {len(self.values)}, "
ValueError: Wrong number of items passed 21, placement implies 18

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\jacksonbe3\Miniconda3\envs\dlc22\lib\site-packages\deeplabcut\gui\refine_tracklets.py", line 318, in create_tracks
    n_tracks=self.ntracks.GetValue(),
  File "C:\Users\jacksonbe3\Miniconda3\envs\dlc22\lib\site-packages\deeplabcut\refine_training_dataset\stitch.py", line 1052, in stitch_tracklets
    stitcher.write_tracks(output_name, animal_names)
  File "C:\Users\jacksonbe3\Miniconda3\envs\dlc22\lib\site-packages\deeplabcut\refine_training_dataset\stitch.py", line 866, in write_tracks
    df = self.format_df(animal_names)
  File "C:\Users\jacksonbe3\Miniconda3\envs\dlc22\lib\site-packages\deeplabcut\refine_training_dataset\stitch.py", line 852, in format_df
    df = pd.DataFrame(data, columns=columns, index=inds)
  File "C:\Users\jacksonbe3\Miniconda3\envs\dlc22\lib\site-packages\pandas\core\frame.py", line 558, in __init__
    mgr = init_ndarray(data, index, columns, dtype=dtype, copy=copy)
  File "C:\Users\jacksonbe3\Miniconda3\envs\dlc22\lib\site-packages\pandas\core\internals\construction.py", line 238, in init_ndarray
    return create_block_manager_from_blocks(block_values, [columns, index])
  File "C:\Users\jacksonbe3\Miniconda3\envs\dlc22\lib\site-packages\pandas\core\internals\managers.py", line 1687, in create_block_manager_from_blocks
    raise construction_error(tot_items, blocks[0].shape[1:], axes, e)
ValueError: Shape of passed values is (158, 21), indices imply (158, 18)
```

</details>


So I don't think the GUI properly supports tracking a single animal in maDLC. 

However, it does work if run in ipython:

```
deeplabcut.stitch_tracklets(config_path, pickle_path, 
                         n_tracks=1,
                        max_gap = 3,
                        min_length=3,
                        )
```

then the refine tracklets GUI shows just `ind1`.

In other words, this is what I think is happening based on my tests:

1.  the OP is setting `Number of animals` to 1, clicking create tracks
2. the 1 reverts to a 2 based on GUI constraint
3. 2 individuals are created in the h5
4. the OP can set `Number of animals` back to 1, then clicks `Refine tracklets`
5. but the GUI loads the h5 with 2 individuals (and the GUI reverts back to 2).
